apps/vaporgui/PImportDataButton.cpp

```diff
@@ -0,0 +1,50 @@
+#include "PImportDataButton.h"
+#include "MainForm.h"
+#include "VHBoxWidget.h"
+#include "VLabel.h"
+#include "PButton.h"
+#include "DatasetTypeLookup.h"
+#include "vapor/ControlExecutive.h"
+#include "vapor/GUIStateParams.h"
+#include "vapor/FileUtils.h"
+
+#include <QFileDialog>
+
+PImportDataButton::PImportDataButton(VAPoR::ControlExec* ce, MainForm *mf) : PWidget("", _hBox = new VHBoxWidget()), _ce(ce), _mf(mf) {
```

Comment:
Please don't pass around main form.

Reply:
Will be removed in 3.9.5 per meeting disuccsion.